libs/feature/record/src/lib/ign-api-dl/ign-api-dl.component.ts

```diff
@@ -0,0 +1,249 @@
+        const params = {
+          zone: zone,
+          format: format,
+          editionDate: editionDate,
+          crs: crs,
+          page: page,
+        } // initialisation des paramètres de filtres
+        for (const [key, value] of Object.entries(params)) {
+          if (value && value !== 'null') {
+            url.searchParams.set(key, String(value))
+          } else {
+            url.searchParams.delete(key)
+          }
+        }
+        outputUrl = url.toString()
+      } else {
+        console.error('erreur apibaseUrl null')
+      }
+      return outputUrl
+    })
+    // startWith(() => this.apiBaseUrl)
+  )
+
+  listFilteredProduct$ = this.apiQueryUrl$.pipe(
+    mergeMap((url) => {
+      console.log(url)
+
+      return this.getFilteredProduct$(url).pipe(
+        map((response) => response['entry'])
+        // startWith([])
+      )
+    })
+  )
+
+  pageMax$ = this.apiQueryUrl$.pipe(
+    mergeMap((url) => {
+      return this.getFilteredProduct$(url).pipe(
+        map((response) =>
+          Math.ceil(response['totalentries'] / Number(this.initialLimit))
+        )
+      )
+    })
+  )
+
+  getFilteredProduct$(url): Observable<any> {
+    return this.http.get(url)
+  }
+
+  getLinkFormat(produit): string {
+    return produit['format'][0]['label']
+  }
+
+  setEditionDate(value: string) {
+    if (value.match(/[0-9]{4}-[0-1][0-9]-[0-3][0-9]/)) {
+      this.editionDate$.next(value)
+      this.resetPage()
+    }
+  }
+
+  setZone(value: string) {
+    if (this.bucketPromisesZone.map((choice) => choice.value).includes(value)) {
+      this.zone$.next(value)
+      this.resetPage()
+    }
+  }
+
+  setCrs(value: string) {
+    if (this.bucketPromisesCrs.map((choice) => choice.value).includes(value)) {
+      this.crs$.next(value)
+      this.resetPage()
+    }
+  }
+
+  setFormat(value: string) {
+    if (
+      this.bucketPromisesFormat.map((choice) => choice.value).includes(value)
+    ) {
+      this.format$.next(value)
+      this.resetPage()
+    }
+  }
+
+  resetUrl() {
+    // this.offset$.next(DEFAULT_PARAMS.OFFSET)
+    this.zone$.next('null')
+    this.format$.next('null')
+    this.crs$.next('null')
+    this.page$.next(1)
+  }
+  moreResult(): void {
+    this.page$.next(this.page$.value + 1)
+  }
+
+  lessResult(): void {
+    this.page$.next(this.page$.value - 1)
+  }
+
+  resetPage(): void {
+    this.page$.next(1)
+  }
+
+  async getCapabilities() {
+    let page = 0
+    let choicesTest = undefined
+    let pageCount = 1
+
+    while (choicesTest === undefined && pageCount > page) {
+      const response = await axios.get(
+        this.url.concat(`&limit=200&page=${page}`)
+      )
+
+      choicesTest = response.data.entry.filter(
+        (element) => element['id'] == this.apiBaseUrl
+      )[0]
+      page += 1
+      pageCount = response.data.pagecount
+    }
+
+    return choicesTest
+  }
+  async getFields() {
+    this.choices = await this.getCapabilities()
```

Comment:
I see a lot of repetition in this part of the code. But maybe I don't understand the logic behind it.
@jahow could we simplify this?